### PR TITLE
Add CUDA 12.5 and update 12.4 builds

### LINKED ISSUE #1264: Regarding the issue of bitsandbytes package supporting cuda12.4 version
### System Info

cuda12.4,python3.11,bitsandbytes-0.43.1

### Reproduction

import torch
from transformers import AutoTokenizer, AutoModelForCausalLM, Trainer, TrainingArguments
from peft import get_peft_model, LoraConfig
from datasets import load_dataset

model_name = "bigscience/bloom-560m"  
tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModelForCausalLM.from_pretrained(model_name)

dataset = load_dataset("imdb")

def preprocess_function(examples):
    return tokenizer(examples['text'], truncation=True, padding='max_length', max_length=512)

tokenized_datasets = dataset.map(preprocess_function, batched=True)

peft_config = LoraConfig(
    task_type="CAUSAL_LM", 
    r=8, 
    lora_alpha=32, 
    lora_dropout=0.1, 
)

model = get_peft_model(model, peft_config)

### Expected behavior

Dear experts, when running the peft fine-tuning program, execute the following code:
model = get_peft_model(model, peft_config)
report errors:
Could not find the bitsandbytes CUDA binary at WindowsPath('D:/Users/1/anaconda3/Lib/site-packages/bitsandbytes/libbitsandbytes_cuda124.dll')
The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU quantization are unavailable.
The latest bitsandbytes package has been installed, version bitsandbytes-0.43.1
I also checked on GitHub, and the latest version supports CUDA 12.4, but
D:/Users/1/anaconda3/Lib/site-packages/bitsandbytes/
There is still no libbitsandbytes_cuda124.dll in the directory, but there are 117 to 123
Please answer my questions, Gods and Masters. Thank you very much

### PR DESCRIPTION
**Pull request by matthewdouglas**
This PR adds CUDA 12.5.0 builds for both Windows and Linux. Additionally, CUDA 12.4.0 is bumped to 12.4.1 and is now enabled for Windows.

Fixes #1264 

@Titus-von-Koeller

### CODE REVIEW

**Review by Titus-von-Koeller — APPROVED** [review]

**Comment by FurkanGozukara** [comment]
i am still getting this error how to fix?

cuda 12.4

**Comment by wangyan-freedom** [comment]
i am still getting this error, with bitsandbytes version of 0.45.0, for cuda 12.4, how to solve it.
